LiteCore/Storage/SQLiteDataFile.cc

```diff
@@ -189,25 +195,41 @@ namespace litecore {
 
     SQLiteDataFile::~SQLiteDataFile() { close(); }
 
+    // A lot of this logic can be turned into a reusable function later if needed
+    // for more extensions.
     static void LoadVectorSearchExtension(sqlite3* sqlite) {
 #ifdef COUCHBASE_ENTERPRISE
+        static const char* extensionName = "CouchbaseLiteVectorSearch";
         if ( sExtensionPath.empty() ) return;
 
         // First enable extension loading (for security reasons it's off by default):
         int rc = sqlite3_db_config(sqlite, SQLITE_DBCONFIG_ENABLE_LOAD_EXTENSION, 1, NULL);
         if ( rc != SQLITE_OK ) {
-            LogToAt(DBLog, Warning, "Unable to enable SQLite extension loading: err %d", rc);
-            return;
+            LogToAt(DBLog, Error, "Unable to enable SQLite extension loading: err %d", rc);
```

Comment:
I guess this the style you put forth as an one of the logging wishes, log and throw. But _throw will also log, right? and with the unique custom message, we can easily find the location where the error occurs, or adding __FILE__/__LINE__, right?

Reply:
`_throw` only logs if it is set to log and as for adding file and line, I think it would be more appropriate to get the macro itself to do it somehow (i.e. define the macro such that it takes a file and line number)